api/src/main/java/jakarta/data/repository/By.java

```diff
@@ -97,6 +97,10 @@
      * The special value which indicates the unique identifier field or
      * property. The annotation {@code By(ID)} maps a parameter to the
      * identifier.
+     * <p>
+     * Note that {@code id(this)} is the expression in JPQL for the
+     * unique identifier of an entity with an implicit identification
+     * variable.
      */
-    String ID = "#id";
+    String ID = "id(this)";
```

Comment:
> Should we have this function included before?

WDYM?

Reply:
Checking the Note:

> Note that {@code id(this)} is the expression in JPQL for the
		     * unique identifier of an entity with an implicit identification
		     * variable.

It is a JPQL function and not a Jakarta Data Query Language function.

I remember that you taught us that the specification should be self-contained. That is, the vendor should read only this specification and understand the behavior of Jakarta Data.

We have included this JPQL function right now by note.

Should we also include this function in our query instead of referring to this JPQL query? Or is it acceptable in this case?

Reply:
Yeah well that's why I asked:

> A question I have for you guys is: should we add `id(this)` and `version(this)` as legal expressions to JPQL?

It's not _critical_ in this case (or at least not yet) because right now this is just a magical constant string value.